ui/app/src/state/reducers/preview.ts

```diff
@@ -227,7 +242,32 @@ const reducer = createReducer<GlobalState['preview']>({
       nextState = { ...nextState, currentUrl: payload.nextUrl };
     }
     return nextState;
-  }
+  },
+  [FETCH_ASSETS_PANEL_ITEMS]: (state, { payload: query }: { payload: ElasticParams }) => ({
+    ...state,
+    assets: { ...state.assets, isFetching: true, query: { ...state.assets.query, ...query } }
+  }),
+  [FETCH_ASSETS_PANEL_ITEMS_COMPLETE]: (state, { payload: searchResult }: { payload: SearchResult }) => {
+    let itemsLookupTable = createLookupTable<MediaItem>(searchResult.items, 'path');
+    let pageNumber = Math.ceil(state.assets.query.offset / state.assets.query.limit);
```

Comment:
We’re having this calculation in many places. I’m thinking we can add the computed value to the state and just use. (i.e. `currentPage`)